wgengine/netstack/netstack_test.go

```diff
@@ -259,6 +259,13 @@ func TestShouldHandlePing(t *testing.T) {
 	}
 }
 
+// looksLikeATailscaleSelfAddress reports whether addr looks like
+// a Tailscale self address, for tests.
+func looksLikeATailscaleSelfAddress(addr netip.Addr) bool {
+	return addr.Is4() && tsaddr.IsTailscaleIP(addr) ||
```

Comment:
As written, an IPv6 Tailscale non-4via6 address will return false. I guess I'd expect this to be: `tsaddr.IsTailscaleIP(addr) || addr.Is6() && tsaddr.Tailscale4To6Range().Contains(addr)`

Reply:
I don't think it will?

Looks right to me: https://go.dev/play/p/FhaV1vqZxqx

The confusion is likely `tsaddr.IsTailscaleIP`: it's not "Is a Tailscale self IP address", but:

```go
// IsTailscaleIP reports whether ip is an IP address in a range that
// Tailscale assigns from.
func IsTailscaleIP(ip netip.Addr) bool {
	if ip.Is4() {
		return CGNATRange().Contains(ip) && !ChromeOSVMRange().Contains(ip)
	}
	return TailscaleULARange().Contains(ip)
}

// TailscaleULARange returns the IPv6 Unique Local Address range that
// is the superset range that Tailscale assigns out of.
func TailscaleULARange() netip.Prefix {
	tsUlaRange.Do(func() { mustPrefix(&tsUlaRange.v, "fd7a:115c:a1e0::/48") })
	return tsUlaRange.v
}
```

.... the whole ULA /48 range, not just the bit we used to translate CGNAT v4s to v6.

Reply:
Oh: 4To6 and 4Via6 are two different things.

I interpreted `tsaddr.Tailscale4To6Range().Contains(addr)` as checking for a 4via6 address and didn't see how a regular Tailscale IPv6 address was handled.